### PR TITLE
Update 1upkeyboards/super16 with code from vendor

### PR DESCRIPTION
**Pull request by jeffmhastings**
## Description

I noticed that the 1upkeyboards/super16 keyboard was out of sync with the firmware provided by the vendor (http://1upkeyboards.com/super16.zip). I did my best to merge the changes from there in with the code that had been committed here previously.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

FYI @mechmerlin

### CODE REVIEW

File: keyboards/1upkeyboards/super16/readme.md [hunk 1]
```diff
@@ -2,12 +2,15 @@
 
 A 4x4 macropad with RGB underglow.
 
-Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)  
-Hardware Supported: Super 16 Macropad  
+Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/readme.md]: Please revert the changes in this file, the deletion of spaces distorts the markdown rendering.

File: keyboards/1upkeyboards/super16/config.h [hunk 2]
```diff
@@ -21,10 +21,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x0000
+#define PRODUCT_ID      0x2010
 #define DEVICE_VER      0x0001
-#define MANUFACTURER    1upkeyboards
-#define PRODUCT         super16
+#define MANUFACTURER    1up Keyboards
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/config.h]: I have been working with 1up to get these boards supported by QMK. I've kept the manufacturer name 1upkeyboards across all of them. If you want to make this change, you'll have to change all the boards in 1upkeyboards.

File: keyboards/1upkeyboards/super16/config.h [hunk 3]
```diff
@@ -21,10 +21,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#define PRODUCT         Super16
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/config.h]: I've actually been trying to keep these all lowercase.

File: keyboards/1upkeyboards/super16/keymaps/default/keymap.c [hunk 4]
```diff
@@ -17,35 +17,23 @@
...
+    RGB_MOD,    KC_1,    KC_U,    KC_P,
+    RGB_TOG,    KC_1,    KC_U,    KC_P,
+    RGB_MOD,    KC_1,    KC_U,    KC_P
   ),
 };
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
   switch (keycode) {
-    case QMKBEST:
+    case UP_URL:
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/keymaps/default/keymap.c]: We're actually not using this in the keymap, we can just remove this.

File: keyboards/1upkeyboards/super16/rules.mk [hunk 5]
```diff
@@ -62,22 +59,20 @@ BOOTLOADER = caterina
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration(+1000)
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration(+1000)
 MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
 EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
-CONSOLE_ENABLE = yes        # Console for debug(+400)
-COMMAND_ENABLE = yes        # Commands for debug and configuration
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/rules.mk]: We have plenty of space left on the eeprom to keep these as yes, are you encountering issues?

File: keyboards/1upkeyboards/super16/rules.mk [hunk 6]
```diff
@@ -62,22 +59,20 @@ BOOTLOADER = caterina
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
```
Comment (mechmerlin) [inline keyboards/1upkeyboards/super16/rules.mk]: We have plenty of space left on the eeprom to keep these as yes, are you encountering issues?

File: keyboards/1upkeyboards/super16/rules.mk (hunk 5 above)
Comment (jeffmhastings) [reply keyboards/1upkeyboards/super16/rules.mk]: Yeah, with those on I was getting errors:

```
Making 1upkeyboards/super16 with keymap default                                                        [ERRORS]

 * The firmware is too large! 30798/28672 (2126 bytes over)
```

I think this is due to the addition of the `#define RGBLIGHT_ANIMATIONS` potentially?

Comment (mechmerlin) [reply keyboards/1upkeyboards/super16/rules.mk]: Probably, you can try adding in optimization flags

```
EXTRAFLAGS += -flto
```

Comment (drashna) [reply keyboards/1upkeyboards/super16/rules.mk]: Yeah, animations take up a lot of space. 

Also, disabling macros/functions may be a good idea, too.

Comment (jeffmhastings) [reply keyboards/1upkeyboards/super16/rules.mk]: With those flags it's still getting really close:

```
Making 1upkeyboards/super16 with keymap default                                                        [WARNINGS]

 * The firmware size is approaching the maximum - 28190/28672 (482 bytes free)
```
 
I'm open to whatever you think is best here.

Comment (mechmerlin) [reply keyboards/1upkeyboards/super16/rules.mk]: Alright that's troubling. When rgb matrix is implemented this is going to get even bigger. Let's stick with turning them off for now.

File: keyboards/1upkeyboards/super16/rules.mk [hunk 7]
```diff
@@ -62,22 +59,20 @@ BOOTLOADER = caterina
...
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
 MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
-HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
-
-LAYOUTS = ortho_4x4
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs (+400)
```
Comment (zvecr) [inline keyboards/1upkeyboards/super16/rules.mk]: Community support was removed, while `LAYOUT_numpad_4x4` was added, was this intentional?
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs (+400)
LAYOUTS = ortho_4x4 numpad_4x4
```

**Comment by drashna** [comment]
Thanks!
